.github/workflows/update-pyproject-version.yml

```diff
@@ -0,0 +1,75 @@
+name: Update pyproject.toml Version
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  update-pyproject-toml:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for all branches and tags
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Check write permissions
+        id: check_permissions
+        run: |
+          if [ "$(jq -r '.permissions.contents' < "$GITHUB_EVENT_PATH")" != "write" ]; then
+            echo "Write permissions are required for this action to run."
+            exit 0
+          fi
+
+      - name: Get latest release tag
+        id: get_latest_tag
+        run: |
+          git fetch --tags
+          LATEST_TAG=$(git describe --tags $(git rev-list --tags --max-count=1))
+          if [ -n "$LATEST_TAG" ]; then
+            echo "::set-output name=latest_tag::$LATEST_TAG"
+          else
+            echo "No release tags found."
+            exit 0
+          fi
+
+      - name: Check if version needs update
+        id: check_version
+        run: |
+          CURRENT_VERSION=$(python -c "import toml; print(toml.load('pyproject.toml')['tool']['poetry']['version'])")
```

Comment:
Maybe we can use `poetry version -s` to get the version number, and use `poetry version x.y.z` to specify the version number, it's more elegant!

Reply:
Imo, the releases page's latest tag should steer the version number, not "some" value in poetry's file, which clearly got forgotten in the past weeks. 😁

Reply:
Oh, having said that, it's dawning on me... how to introduce a new tag via code for a _new_ release would require the poetry version to change and go to a tag.
Hmm... then the Action in this file indeed needs to work differently.

Reply:
@Shimada666 Since the action is already reading both the version from Poetry and the latest tag (if one exists), maybe it works with this process:
- it already has both versions read
- if they are the same, no further action, action is done
- if the current tag's semantic version is higher, the version in Poetry is updated (as the action does right now)
- **_new step:_** if Poetry has a newer version, a new tag needs to be created for it - Todo!

The last point is not yet in, I'll have to look into this.

What do you think about the flow above?

Reply:
@tobitege 
Sounds good! Maybe we don't need to consider the last step. We can assume the version in pyproject.toml is only updated by this GitHub action.

Reply:
So maybe we just need to do one simple thing: after manually creating a GitHub release, have the GitHub action sync the generated tag version to pyproject.toml.

Reply:
> @tobitege Sounds good! Maybe we don't need to consider the last step. We can assume the version in pyproject.toml is only updated by this GitHub action.

That is not yet clear to me and certainly an issue that @rbren and @neubig and others should decide on (version as chicken and egg problem 🤣).
If a new release should roll out with a PR, it'd be quite handy if the action could take care of the tag.

Reply:
🤣🤣